fiftyone/core/collections.py

```diff
@@ -3243,23 +3390,20 @@ def aggregate(self, aggregations, _attach_frames=True):
 
         Returns:
             an aggregation result or list of aggregation results corresponding
-            to the input aggregations
+            to the input aggregation(s)
         """
         scalar_result, aggregations, facets = self._build_aggregation(
             aggregations
         )
-        if len(aggregations) == 0:
+        if not aggregations:
             return []
 
-        # pylint: disable=no-member
         pipeline = self._pipeline(
             pipeline=facets, attach_frames=_attach_frames
         )
-        try:
-            # pylint: disable=no-member
-            result = next(self._dataset._sample_collection.aggregate(pipeline))
-        except StopIteration:
-            pass
+
+        result = self._dataset._sample_collection.aggregate(pipeline)
```

Comment:
@benjaminpkane I couldn't quite figure out why the `try-except` was here previously. It seems like things wouldn't work if an exception was actually caught there anyway?  It seems that the `if not aggregations` check above is all that is required.

I wonder if this was something to do with handling empty datasets in the App.

The reason I was mucking around here is that I wanted errors to occur if someone runs an invalid aggregation on their data. Ideally the app's aggregations wouldn't rely on error catching for empty edge cases.

Reply:
Yes, I suppose `result` would not be declared. I think you're right to remove it.